pkg/controlsvc/controlsvc_test.go

```diff
@@ -0,0 +1,647 @@
+			},
+		},
+		{
+			name:          "service listener error",
+			expectedError: "error opening Unix socket: terminated",
+			expectedCalls: func() {
+				mockNetceptor.EXPECT().ListenAndAdvertise(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, errors.New("terminated"))
+			},
+			listeners: map[string]string{
+				"service":    "service",
+				"unixSocket": "",
+				"tcpListen":  "",
+			},
+		},
+		{
+			name:          "no listeners error",
+			expectedError: "no listeners specified",
+			listeners: map[string]string{
+				"service":    "",
+				"unixSocket": "",
+				"tcpListen":  "",
+			},
+		},
+	}
+
+	for _, testCase := range runControlSvcTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			if testCase.expectedCalls != nil {
+				testCase.expectedCalls()
+			}
+			s := controlsvc.New(false, mockNetceptor)
+			s.SetServerUtils(mockUnix)
+			s.SetServerNet(mockNet)
+
+			err := s.RunControlSvc(context.Background(), testCase.listeners["service"], &tls.Config{}, testCase.listeners["unixSocket"], os.FileMode(0o600), testCase.listeners["tcpListen"], &tls.Config{})
+
+			if err == nil || err.Error() != testCase.expectedError {
+				t.Errorf("expected error %s, got %v", testCase.expectedError, err)
+			}
+		})
+	}
+}
+
+func TestSockControlRemoteAddr(t *testing.T) {
+	ctrl := gomock.NewController(t)
+
+	mockCon := mock_controlsvc.NewMockConn(ctrl)
+	mockAddr := mock_controlsvc.NewMockAddr(ctrl)
+	mockUtil := mock_controlsvc.NewMockUtiler(ctrl)
+	mockCopier := mock_controlsvc.NewMockCopier(ctrl)
+	sockControl := controlsvc.NewSockControl(mockCon, mockUtil, mockCopier)
+
+	localhost := "127.0.0.1"
+
+	mockCon.EXPECT().RemoteAddr().Return(mockAddr)
+	mockAddr.EXPECT().String().Return(localhost)
+	remoteAddr := sockControl.RemoteAddr()
+
+	if remoteAddr.String() != localhost {
+		t.Errorf("expected: %s, received: %s", localhost, remoteAddr)
+	}
+}
+
+func TestSockControlWriteMessage(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockCon := mock_controlsvc.NewMockConn(ctrl)
+	mockUtil := mock_controlsvc.NewMockUtiler(ctrl)
+	mockCopier := mock_controlsvc.NewMockCopier(ctrl)
+
+	sockControl := controlsvc.NewSockControl(mockCon, mockUtil, mockCopier)
+
+	writeMessageTestCases := []struct {
+		name          string
+		message       string
+		expectedCalls func()
+	}{
+		{
+			name:    "without message",
+			message: "",
+		},
+		{
+			name:    "with message",
+			message: "message",
+			expectedCalls: func() {
+				mockCon.EXPECT().Write(gomock.Any()).Return(0, errors.New("cannot write message"))
+			},
+		},
+	}
+
+	for _, testCase := range writeMessageTestCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			if testCase.expectedCalls != nil {
+				testCase.expectedCalls()
+			}
+			err := sockControl.WriteMessage(testCase.message)
+
+			if testCase.message == "" && err != nil {
+				t.Errorf("should be nil")
+			}
+			if testCase.message != "" && err.Error() != "cannot write message" {
+				t.Errorf("%s %s", testCase.name, err)
```

Comment:
The testCase.name is already printed as part of the `t.Run` command above, please rewrite this error :smile: